feat: UI/UX improvements - moved search box to the top

prakashrx opened the pull request:
Move the search input to the top so that users can easily and quickly search for abbrev definitions

Closes #86

### What new changes did you make? Tick all applicable boxes
- [ ] Added new abbreviation
- [ ] Fixed something in the source code
- [x] Added a new feature
- [ ] Fixed the docs (README.md, CONTRIBTUING.md, etc)

<!--Give a brief outline of changes you made. If you added slangs, which ones? -->
### Changes
 - Currently, users have to scroll down in order to search for abbreviations. With this change, the search form is moved to the top along with the banner. Makes it easier for users to search from a UX perspective.
 - I have kept the "Found something odd? Create a github issue." section in the footer as I felt it would be odd to have this in the hero section.
 - Fixed formatting/lint issue within form.jsx . (Taken care by VS code format on save). Functionally, the only change in this file is some additional tailwind classes to center the form and add some margin.
 
<!--Optional, but advised -->
### Share a screeshot of new changes
![image](https://user-images.githubusercontent.com/2094439/195000162-3a36a97f-08ab-4620-8c5c-320b618fdf8c.png)

Mobile
![image](https://user-images.githubusercontent.com/2094439/195000406-e80b1072-5c01-48a1-b527-c9fb48d410e6.png)

Footer
![image](https://user-images.githubusercontent.com/2094439/195003353-7bb3d9f7-bea7-4897-b9cd-029fe53cf5e3.png)

adropofilm reviewed (changes requested):
Great work! Minor change required however. Please move the circled text to the right of the IYKYK text:

**Actual Behavior:**
![Screen Shot 2022-10-11 at 1 45 16 AM](https://user-images.githubusercontent.com/9134674/195015620-6cd03c1f-8d8a-4859-9136-0c3453d693e2.jpg)

**Expected Behavior**
![Screen Shot 2022-10-11 at 1 47 06 AM](https://user-images.githubusercontent.com/9134674/195015799-22ae2909-9f44-4909-86da-77d5347143ce.jpg)

prakashrx commented:
@adropofilm I missed that. fixed that few other things that I noticed in small screen layouts.

- moved "if you know, you know" text back to its original position
- fixed position of "Find an abbreve(tion)" in small screens
- fixed margins on small screens. too much free space at element tops

Njong392 commented:
Since the form has been moved closer to the top in small screens, I think we can remove the 'Find an abbreviation' button completely.

And on mobile, the 'Get Started' button should be removed from the nav bar too.

I also have another change to propose. I think the form component would look better (on desktop) if it was below the chain of animations. That is, the moving animation with abbreviation boxes. How does this sound? @prakashrx

prakashrx commented:
@Njong392 here is a screenshot of proposed changes. Looks good.

![image](https://user-images.githubusercontent.com/2094439/195032356-67179f39-f436-4d0c-ad62-7cdc1ae36019.png)

In, small screens, removed "Find an abbreve(tion)".  
'Get Started' button is already hidden in small screens. Is it not?

![image](https://user-images.githubusercontent.com/2094439/195032640-6f090ec6-d770-40fc-8eac-fc394cdfd571.png)

Njong392 commented:
I meant the 'Get Started' on the navbar for desktop. Sorry, i mistyped it.

I look at the main heading and the animations as one, and that's why I thought putting the search bar below would look better.

prakashrx commented:
Cool. I have made the modifications
- move search box below the animation
- Removed "Find an abbreve(tion)" text in small screens
- Removed "Get Started" button in desktop/small screens

Njong392 commented:
It looks great! But some things have been kind of misaligned. The error messages have been kind of shifted to the right. 

![Screenshot from 2022-10-11 13-35-46](https://user-images.githubusercontent.com/81039882/195092522-848709b4-fe86-4023-a71c-5b7bd8b4dafc.png)
![Screenshot from 2022-10-11 13-35-58](https://user-images.githubusercontent.com/81039882/195092579-2e5edea4-c414-4a7d-a145-3e26675488df.png)

prakashrx commented:
My bad. I removed the `text-center` wrapper div around the form.

adropofilm commented:
Looks good to me! :) Thank you for the revisions @prakashrx
